src/main/webapp/js/community-page-loader.js

```diff
@@ -1,33 +1,57 @@
+const React = window.React;
+const ReactDOM = window.ReactDOM;
+const h = React.createElement;
+const root = document.getElementById("list");
 
-/** Fetches users and adds them to the page. */
-function fetchUserList(){
-const url = '/user-list';
-fetch(url).then((response) => {
-  return response.json();
-}).then((users) => {
-  const list = document.getElementById('list');
-  list.innerHTML = '';
+class Community extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {};
+    this.fetchUserList = this.fetchUserList.bind(this);
+  }
+  /** Adds list of community users to community.html as a list element */
+  render() {
+    const { Users } = this.state;
+    if (Users == null) {
+      return h("li", null, "loading...");
+    }
+    const children = [];
+    for (const user of Users) {
+      children.push(h(CommunityUser, { email: user }));
+    }
+    return h("li", null, children);
+  }
 
-  users.forEach((user) => {
-    const userListItem = buildUserListItem(user);
-    list.appendChild(userListItem);
-    });
-  });
-}
-/**
-* Builds a list element that contains a link to a user page, e.g.
-* <li><a href="/user-page.html?user=test@example.com">test@example.com</a></li>
-*/
-function buildUserListItem(user){
-  const userLink = document.createElement('a');
-  userLink.setAttribute('href', '/user-page.html?user=' + user);
-  userLink.appendChild(document.createTextNode(user));
-  const userListItem = document.createElement('li');
-  userListItem.appendChild(userLink);
-  return userListItem;
+  /** Fetches users from /user-list */
+  fetchUserList() {
+    const url = "/user-list";
+    fetch(url)
+      .then(response => {
+        return response.json();
+      })
+      .then(users => {
+        this.setState({ Users: users });
+      });
+  }
+
+  componentDidMount() {
+    this.fetchUserList();
+    this.timer = window.setInterval(this.fetchUserList, 500);
```

Comment:
Please remove this timer for now; I don't want to spam our AppEngine server when we deploy.

Reply:
If i remove the timer, I need to also remove the "compoentWillUnmount" function right?
I did this and Its not updating the page with the users?

Reply:
yes and i think we figured that out in person